x/storage/keeper/keeper.go

```diff
@@ -585,14 +601,22 @@ func (k Keeper) CreateObject(
 		creator = operator
 	}
 
-	// check approval
-	if opts.PrimarySpApproval.ExpiredHeight < uint64(ctx.BlockHeight()) {
+	// TODO: Select the correct hard fork version to update vgf
+	if !ctx.IsUpgraded(upgradetypes.Pawnee) && opts.PrimarySpApproval.ExpiredHeight < uint64(ctx.BlockHeight()) {
 		return sdkmath.ZeroUint(), errors.Wrapf(types.ErrInvalidApproval, "The approval of sp is expired.")
 	}
 
-	err = k.VerifySPAndSignature(ctx, sp, opts.ApprovalMsgBytes, opts.PrimarySpApproval.Sig, operator)
-	if err != nil {
-		return sdkmath.ZeroUint(), err
+	// TODO: Select the correct hard fork version to update vgf
+	if !ctx.IsUpgraded(upgradetypes.Pawnee) {
```

Comment:
Can we move the `ExpiredHeight` check into this branch?

Reply:
fixed